### PR TITLE
:hammer: driver / form / API

### PR DESCRIPTION
**Pull request by fabienjuif**
### TODOs
 - [x] New tests (addOrUpdate)
 - [x] Test that `set` removed older keys
 - [x] TODO in code (rename snapshots)
 - [x] Update the breaking change section (markdown)
 - [x] Document `find`

### CODE REVIEW

File: packages/drivers/form/README.md [hunk 1]
```diff
@@ -53,6 +53,40 @@ const mapStore = inject((store, { formName, field }, drivers) => {
 export default mapStore(Component)
 ```
 
+## API
+### Local to a form
+These functions are available on form that is returned by this call: `driver.form(formName)` where `formName` is you form name.
```
Comment (fabienjuif) [inline packages/drivers/form/README.md]:
```suggestion
These functions are availables on form that is returned by this call: `driver.form(formName)` where `formName` is you form name.
```

Comment (guillaumecrespel) [inline packages/drivers/form/README.md]:
```suggestion
These functions are available on form that is returned by this call: `driver.form(formName)` where `formName` is your form name.
```

File: packages/drivers/form/README.md [hunk 2]
```diff
@@ -53,6 +53,40 @@ const mapStore = inject((store, { formName, field }, drivers) => {
...
+| `update(fieldName)(value)` | update the value of the given fieldName to the `formName` form |
+| `addOrUpdate(<object>)` | add or update given values to the `formName` form |
+| `setErrors(<object>)` | set errors to the `formName` form, and remove the old ones |
+| `addOrUpdateErrors(<object>)` | add or update errors to the `formName` form |
+| `resetErrors()` | remove all errors to the `formName` form |
+| `reset()` | remove all errors and values to the `formName` form |
+| --- | --- |
+| `exists()` | test that the `formName` form exists |
+| `get()` | retrieve the key/value object from the `formName` form, all fields are retrieved |
+| `get(<string>)` | retrieve the value of the given field name from the `formName` form |
```
Comment (guillaumecrespel) [inline packages/drivers/form/README.md]:
```suggestion
| `get(fieldName)` | retrieve the value of the given `fieldName` from the `formName` form |
```

File: packages/drivers/form/README.md [hunk 3]
```diff
@@ -53,6 +53,40 @@ const mapStore = inject((store, { formName, field }, drivers) => {
...
+| `getErrors()` | retrieve the key/value error object from the `formName` form |
+| `getErrors(<string>)` | retrieve error of the given field name from the `formName` form |
```
Comment (guillaumecrespel) [inline packages/drivers/form/README.md]:
```suggestion
| `getErrors(fieldName)` | retrieve error of the given `fieldName` from the `formName` form |
```

File: packages/drivers/form/README.md [hunk 4]
```diff
@@ -53,6 +53,40 @@ const mapStore = inject((store, { formName, field }, drivers) => {
...
+
+### Batched to all given forms at once
+
+These functions are used right into the `driver.form` field without giving a name to it.
```
Comment (guillaumecrespel) [inline packages/drivers/form/README.md]:
```suggestion
These functions are used right into the `drivers.form` field without giving a name to it.
```

File: packages/drivers/form/README.md (hunk 1 above)
Comment (guillaumecrespel) [inline packages/drivers/form/README.md]:
```suggestion
These functions are available on form that is returned by this call: `drivers.form(formName)` where `formName` is you form name.
```

File: packages/drivers/form/README.md [hunk 5]
```diff
@@ -53,6 +53,40 @@ const mapStore = inject((store, { formName, field }, drivers) => {
...
+
+| function | description |
+| --- | --- |
+| `set([{ name: <string>, values: <object> }, ...])` | set `values` to the form identified by `name`, older values are removed |
+| `addOrUpdate([{ name: <string>, values: <object> }, ...])` | add or update `values` to the form identified by `name` |
+| `resetErrors([<string>, ...])` | reset errors for all the given form names |
```
Comment (guillaumecrespel) [inline packages/drivers/form/README.md]:
```suggestion
| `resetErrors([formName, ...])` | reset errors for all the given form names |
```

File: packages/drivers/form/README.md [hunk 6]
```diff
@@ -53,6 +53,40 @@ const mapStore = inject((store, { formName, field }, drivers) => {
...
+| `reset([<string>, ...])` | reset errors and values for all the given form names |
```
Comment (guillaumecrespel) [inline packages/drivers/form/README.md]:
```suggestion
| `reset([formName, ...])` | reset errors and values for all the given form names |
```

File: packages/drivers/form/README.md [hunk 7]
```diff
@@ -53,6 +53,40 @@ const mapStore = inject((store, { formName, field }, drivers) => {
 export default mapStore(Component)
 ```
 
+## API
+### Local to a form
+These functions are availables on form that is returned by this call: `driver.form(formName)` where `formName` is you form name.
```
Comment (fabienjuif) [inline packages/drivers/form/README.md]:
```suggestion
These functions are availables on form that is returned by this call: `driver.form(formName)` where `formName` is your form name.
```

File: packages/drivers/form/README.md (hunk 2 above)
Comment (fabienjuif) [reply packages/drivers/form/README.md]: I let it that way for consistency

File: packages/drivers/form/README.md (hunk 1 above)
Comment (fabienjuif) [reply packages/drivers/form/README.md]: I don't know why I can't use this suggestion, so I fixed it manually

File: packages/drivers/form/README.md [hunk 8]
```diff
@@ -53,6 +53,40 @@ const mapStore = inject((store, { formName, field }, drivers) => {
 export default mapStore(Component)
 ```
 
+## API
+### Local to a form
+These functions are availables on form that is returned by this call: `driver.form(formName)` where `formName` is your form name.
```
Comment (fabienjuif) [inline packages/drivers/form/README.md]:
```suggestion
These functions are availables on form that is returned by this call: `drivers.form(formName)` where `formName` is your form name.
```

File: packages/drivers/form/README.md (hunk 1 above)
Comment (fabienjuif) [reply packages/drivers/form/README.md]: Same, I can't use this suggestion so I fixed it manually

**Pull request merged**